pkg/release/chart.go

```diff
@@ -204,10 +203,29 @@ func (rel *config) getChartRepoEntryFromIndex(u, repositoryCache string) (*repo.
 		return nil, fmt.Errorf("no cached repo found: %w", err)
 	}
 
+	rf, err := repo.LoadFile(rel.getDownloader().RepositoryConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load repository file: %w", err)
+	}
+	var baseUrl string
+	for _, rc := range rf.Repositories {
+		if rc.Name == repoName {
+			baseUrl = rc.URL
+		}
```

Comment:
@Vigilans What do you think about the break option here?

Reply:
It can break. The referencing helm's source code just breaks the loop by `return`:
https://github.com/helm/helm/blob/47c8cded543c3ef34ad859734dc5c1e84510b5b4/pkg/downloader/chart_downloader.go#L341-L351
```
func pickChartRepositoryConfigByName(name string, cfgs []*repo.Entry) (*repo.Entry, error) {
	for _, rc := range cfgs {
		if rc.Name == name {
			if rc.URL == "" {
				return nil, errors.Errorf("no URL found for repository %s", name)
			}
			return rc, nil
		}
	}
	return nil, errors.Errorf("repo %s not found", name)
}
```